### PR TITLE
Feat/offcanvas

### PR DESCRIPTION
**Pull request by div-Leo**
**What:**
Add `Offcanvas` component
​
**Why:**
We need an `Offcanvas` for this [design](https://app.zeplin.io/project/57fe438f81b870887cc71736/screen/60900157d95163088c092d08).
​
**How:**
Add a new component `Offcanvas` which follows the same logic of the `Modal`. 
It receives the side props ("left" | "right" | "top" | "bottom") and slides in from the relative side.
​
**Media:**
Screenshot 2021-06-10 at 17 57 10

See more in this video on [loom](https://www.loom.com/share/eed122adea8a4c539981bfe6557a6e68)

**Checklist:**

- [x] Ready to be merged

See this [pull-request](https://github.com/freenowtech/wave/pull/114) thread for more info.

### CODE REVIEW

File: src/components/Offcanvas/docs/Offcanvas.mdx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import { StyledSystemLinks } from '../../../docs/StyledSystemLinks'
+
+# Offcanvas
+
+The Offcanvas should only contain content or flows which are related to the current page and not replace pages, therefore
+its complexity level has to be simple. Consider the fullscreen state if you want to focus the users attention to the Offcanvas
+content.
+
+Offcanvas is responsive per default but the width can be adjusted (the default is 600px or 37.5rem). Content on the Offcanvas
+is scrollable and has a space5 bounding box.
```
Comment (div-Leo) [inline src/components/Offcanvas/docs/Offcanvas.mdx:19]: Maybe we need a different text here, do we have some in charge for this particular task?

**Review by rafael-sepeda — APPROVED** [review]

File: src/components/Offcanvas/docs/Offcanvas.mdx (hunk 1 above)
Comment (duranmla) [reply src/components/Offcanvas/docs/Offcanvas.mdx:19]: We need to review this. @agonzalezUX @rafael-sepeda can you read this documentation?
Sadly we yet don't have preview links so can you read this file?

**Review by duranmla — APPROVED** [review]
I think this is ready to go. @div-Leo what is missing that you haven't check the ready for merge on top of the docs review?

**Comment by div-Leo** [comment]
@duranmla It was just for the documentation, but for me, it's ready to go!  woot woot! 💪 🚂

**Pull request merged**